app/controllers/uploads_controller.rb

```diff
@@ -0,0 +1,23 @@
+class UploadsController < ApplicationController
+  def new
+  end
+
+  def create
+    obj = S3_BUCKET.objects[params[:file].original_filename]
+
+    obj.write(file: params[:file], acl: :public_read)
+
+    @upload = Upload.new(url: obj.public_url, name: obj.key)
+
+    if @upload.save
+      redirect_to uploads_path, success: 'File successfully uploaded'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.